src/sentry/models/outbox.py

```diff
@@ -524,28 +521,45 @@ def save(self, **kwds: Any) -> None:  # type: ignore[override]
         metrics.incr("outbox.saved", 1, tags=tags)
         super().save(**kwds)
 
+    def lock_id(self, attrs: Iterable[str]) -> int:
+        # 64 bit integer that roughly encodes a unique, serializable lock identifier
+        return mmh3.hash64(".".join(str(getattr(self, attr)) for attr in attrs))[0]
```

Comment:
Postgres advisory locks are backed by 64 bit keys... unfortunately they aren't scoped in any other way.  We perform a 64 bit hashing of the sharding values to produce one.  That means it is possible to have collision: but that's not awful.  A collision only means that two distinct shards end up blocking and competing against each other, but it is highly unlikely in practice.  It's the price to pay for using session rather than transaction backed locks.